homeassistant/components/sensor/forecast.py

```diff
@@ -129,75 +122,72 @@ def unit_of_measurement(self):
     @property
     def unit_system(self):
         """Return the unit system of this entity."""
-        return self._unit_system
+        return self.forecast_data.unit_system
+
+    def update_unit_of_measurement(self):
+        """Update units based on unit system."""
+        unit_index = {
+            'si': 1,
+            'us': 2,
+            'ca': 3,
+            'uk': 4,
+            'uk2': 5
+        }.get(self.unit_system, None)
+        self._unit_of_measurement = SENSOR_TYPES[self.type][unit_index]
```

Comment:
Good catch. Not sure how I missed that. Probably should just default to `1` for the `si` units.